xyzservices/lib.py

```diff
@@ -28,6 +28,70 @@ def __call__(self, **kwargs):
         new.update(kwargs)
         return new
 
+    def build_url(self, x=None, y=None, z=None, token=None, scale_factor=None):
+        """
+        Build the URL of tiles from the TileProvider object
+
+        Can return URL with placeholders or the final tile URL.
+
+        Attributes
+        ----------
+
+        x, y, z : int
+            tile number
```

Comment:
Those are optional?

Reply:
Yes, for use in folium and similar, you want to keep placeholders so the app can load what it needs. I'll add `(optional)`.